core/src/main/java/org/apache/calcite/rel/core/JoinInfo.java

```diff
@@ -90,7 +98,9 @@ public ImmutableBitSet rightSet() {
     return ImmutableBitSet.of(rightKeys);
   }
 
-  public abstract RexNode getRemaining(RexBuilder rexBuilder);
+  public RexNode getRemaining(RexBuilder rexBuilder) {
+    return RexUtil.composeConjunction(rexBuilder, nonEquiConditions);
+  }
```

Comment:
It's so weird that we pass in `nonEquiConditions` into the constructor and return a `RexNode` here, can we have a better solution ?

Reply:
I agree, but given this is a public API, we can't change it to returning list instead of a RexNode. What we can do is deprecating this method, and introduce another one returning conjunction list.

Reply:
I'm +1 on deprecate this method and introduce another one returning conjunction list.